components/hab/src/license.rs

```diff
@@ -152,19 +172,21 @@ fn writeable_license_path() -> PathBuf {
     license_path(&root_dir)
 }
 
-fn env_var_present() -> Result<bool> {
+fn env_var_present() -> Result<LicenseAcceptance> {
     match env::var(LICENSE_ACCEPT_ENVVAR) {
         Ok(val) => {
             if &val == "accept" {
                 write_license_file()?;
-                Ok(true)
+                Ok(LicenseAcceptance::Accepted)
             } else if &val == "accept-no-persist" {
-                Ok(true)
+                Ok(LicenseAcceptance::Accepted)
+            } else if &val == "deny" {
```

Comment:
Do we need to add the concept of `deny`?

Reply:
> Do we need to add the concept of `deny`?

No, we do not. It makes testing slightly more robust because we do not need to rely on timing out the license acceptance prompt. I also thought about a `no-prompt` option. Do we think `deny` or `no-prompt` could ever be useful outside of our own testing? If not I will remove that logic.

Reply:
Oh I had a feeling it may have been for testing but the connection was not super clear to me. Now I see what you are doing with that. It not be a bad idea to add a comment on the motivation behind deny, but it seems legit to me.

Reply:
Agreed. Addressed in [72e3ba](https://github.com/habitat-sh/habitat/pull/7834/commits/72e3ba8a860d9c78dedf679540b814909bce1672)